syntax = "proto3";

option java_package = "grp.dtop.dtopjava.proto";

import "message.proto";
import "google/protobuf/empty.proto";

service GRPCService {
  rpc GetIP (google.protobuf.Empty) returns (StringMessage);
  rpc GetPort (google.protobuf.Empty) returns (IntegerMessage);
  rpc GetAddr (google.protobuf.Empty) returns (StringMessage);
  rpc Profile (FetchRequestMessage) returns (FetchReplyMessage);
  rpc GetServerStatus (StringArrayMessage) returns (ServerStatusMessage);
}